julearn/pipeline/merger.py

```diff
@@ -0,0 +1,116 @@
+from typing import Dict
+
+from sklearn.model_selection import GridSearchCV, RandomizedSearchCV
+from sklearn.pipeline import Pipeline
+
+from ..utils.logging import raise_error
+from ..utils.typing import EstimatorLike
+from .pipeline_creator import _prepare_hyperparameter_tuning
+from ..prepare import prepare_search_params
+
+
+def merge_pipelines(
+    *pipelines: EstimatorLike, search_params: Dict
+) -> Pipeline:
+    """Merge multiple pipelines into a single one.
+
+    Parameters
+    ----------
+    pipelines : List[EstimatorLike]
+        List of estimators that will be merged.
+    search_params : Dict
+        Dictionary with the search parameters.
+
+    Returns
+    -------
+    merged : BaseSearchCV
+        The merged pipeline as a searcher.
+    """
+
+    # Check that we only merge pipelines and searchers. And if there is a
+    # searcher, they are all of the same kind and match the search params.
+
+    search_params = prepare_search_params(search_params)
+
+    for p in pipelines:
+        if not isinstance(p, (Pipeline, GridSearchCV, RandomizedSearchCV)):
```

Comment:
[Consideration, I will accept the PR even if decided against this]
For now I think this is totally fine as it covers 99% use-cases, but just keep in mind that we also support custom searchers and these might be build to also support multiple dicts.  From looking at the implementation I do not see a good way of changing this as the attribute that hold parameters is called differently like param_grid and param_distribution, but maybe we should document that this is only possible for explicitly supported searchers.

Reply:
Agree. So far we can only merge these two searchers. Also, the "custom searcher" must accept having a list of grids/distributions in order to allow for this feature. If you use a custom searcher, then you need to provide one creator without repeated names.